keras_core/ops/image.py

```diff
@@ -260,6 +260,109 @@ def affine_transform(
         data_format=data_format,
     )
 
+class MapCoordinates(Operation):
+    def __init__(
+        self,
+        order,
+        mode,
+        cval=0.0
+    ):
+        super().__init__()
+        self.order = order
+        self.mode = mode
+        self.cval = cval
+
+    def call(self, image, coordinates):
+        return backend.image.map_coordinates(
+            image,
+            coordinates,
+            order=self.order,
+            mode=self.mode,
+            cval=self.cval,
+        )
+
+    def compute_output_spec(self, image, coordinates):
+        if coordinates.shape[0] != len(image.shape):
+            raise ValueError(
+                "First dim of `coordinates` must be the same as the rank of `image`"
+                f"Received image with shape: {image.shape} and coordinate leading dim of "
+                f"{coordinates.shape[0]}"
+            )
+        if len(coordinates.shape) < 2:
+            raise ValueError(
+                "Invalid coordinates rank: expected at least rank 2."
+                f" Received input with shape: {coordinates.shape}"
+            )
+        return KerasTensor(coordinates.shape[1:], dtype=image.dtype)
+
+
+@keras_core_export("keras_core.ops.image.map_coordinates")
+def map_coordinates(
+    input, coordinates, order, mode="constant", cval=0.0
```

Comment:
The argument names should be consistent with the APIs we already have (like `affine_transform`), e.g. `cval` should be something like `fill_value`